src/reducers/userReducer.tsx

```diff
@@ -0,0 +1,42 @@
+const initialState = {
+  displayName: "",
+  email: "",
+  uid: "",
+  isSignedIn: false,
+};
+
+interface UserState {
+  displayName: string;
+  email: string;
+  uid: string;
+  isSignedIn: boolean;
+}
+
+const userReducer = (
+  state: UserState = initialState,
+  action: {
+    type: string;
+    payload: { displayName: string; email: string; uid: string };
+  }
+) => {
+  switch (action.type) {
+    case "SIGNIN":
+      return {
+        displayName: action.payload.displayName,
+        email: action.payload.email,

```

Comment:
Could you use the spread operator to spread the payload out instead of explicitly assigning each of it's keys?